app/api/github/setup/route.ts

```diff
@@ -0,0 +1,206 @@
+    const params = {
+      owner,
+      repo,
+      path,
+      message,
+      content: Buffer.from(content).toString("base64"),
+      branch,
+      ...(sha ? { sha } : {}),
+    };
+
+    await octokit.rest.repos.createOrUpdateFileContents(params);
+    return { success: true };
+  } catch (error: any) {
+    if (error.status === 409) {
+      throw new Error(`Conflict while updating ${path}. Please try again.`);
+    }
+    if (error.status === 422) {
+      throw new Error(
+        `Validation failed for ${path}. Please check file content and permissions.`,
+      );
+    }
+    if (error.status === 404) {
+      throw new Error(`Path ${path} not found or insufficient permissions.`);
+    }
+    throw error;
+  }
+};
+
+export const POST = async (request: NextRequest) => {
+  const { userId } = auth();
+
+  if (!userId) {
+    return NextResponse.redirect(new URL("/login", request.url));
+  }
+
+  try {
+    const octokit = await getOctokit();
+    const body = await request.json();
+
+    const validatedData = SetupRequestSchema.parse(body) satisfies SetupRequest;
+    const { repoFullName, testPattern, triggers, defaultBranch } =
+      validatedData;
+    const [owner, repo] = repoFullName.split("/");
+
+    try {
+      await octokit.rest.repos.get({
+        owner,
+        repo,
+      });
+    } catch (error) {
+      console.error("Error accessing repository:", error);
+      return NextResponse.json(
+        {
+          error:
+            "Repository not found or not accessible. Please make sure the repository exists and you have the right permissions.",
+        },
+        { status: 404 },
+      );
+    }
+
+    await createOrUpdateRepositoryConfig({
+      owner,
+      repo,
+    });
+
+    const configContent = `import type { ShortestConfig } from "@antiwork/shortest";
+
+export default {
+  headless: false,
+  baseUrl: "http://localhost:3000",
+  testPattern: '${testPattern}',
+  ai: {
+    provider: "anthropic",
+    apiKey: process.env.ANTHROPIC_API_KEY
+  }
+} satisfies ShortestConfig;`;
+
+    const workflowContent = `name: Shortest E2E Tests
+
+on:
+${triggers.onPush ? `  push:\n    branches:\n      - ${defaultBranch}\n` : ""}
+${triggers.onPullRequest ? "  pull_request:\n" : ""}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      ANTHROPIC_API_KEY: \${{ secrets.ANTHROPIC_API_KEY }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '22'
+      - name: Install dependencies
+        run: npm install
+      - name: Run Shortest tests
+        run: npx shortest run`;
+
+    try {
+      // Create or update files serially to avoid conflicts
+      await createOrUpdateFile(
+        octokit,
+        owner,
+        repo,
+        "shortest.config.ts",
+        configContent,
+        "Add Shortest configuration",
+        defaultBranch,
```

Comment:
Generally, `main` branches are protected, and I don't think Shortest should push commits directly to `main`. In a CI setup, changes to `main` may trigger a production deployment, and there is a significant risk allowing a 3rd party app to have that type of access.

Reply:
good point! are we good opening a PR instead?